File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/adapters/FoundLyricsAdapter.java
package com.atomykcoder.atomykplay.adapters;

import static com.atomykcoder.atomykplay.constants.FragmentTags.ADD_LYRICS_FRAGMENT_TAG;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.atomykcoder.atomykplay.R;
import com.atomykcoder.atomykplay.ui.MainActivity;
import com.atomykcoder.atomykplay.adapters.viewHolders.FoundLyricsViewHolder;
import com.atomykcoder.atomykplay.fragments.AddLyricsFragment;

import java.util.ArrayList;

public class FoundLyricsAdapter extends RecyclerView.Adapter<FoundLyricsViewHolder> {
    private final ArrayList<String> titles;
    private final ArrayList<String> sampleLyrics;
    private final ArrayList<String> urls;
    private final MainActivity mainActivity;

    public FoundLyricsAdapter(ArrayList<String> _titles, ArrayList<String> _sampleLyrics, ArrayList<String> _urls, Context _context) {
        titles = _titles;
        sampleLyrics = _sampleLyrics;
        urls = _urls;
        mainActivity = ((MainActivity) _context);
    }

    @NonNull
    @Override
    public FoundLyricsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.found_lyrics_item, parent, false);
        return new FoundLyricsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull FoundLyricsViewHolder holder, int position) {
        holder.song_title.setText(titles.get(position));
        holder.song_sampleLyrics.setText(sampleLyrics.get(position));

        holder.itemView.setOnClickListener(view -> {
            mainActivity.setBottomSheetState();
            AddLyricsFragment fragment = (AddLyricsFragment) mainActivity.getSupportFragmentManager().findFragmentByTag(ADD_LYRICS_FRAGMENT_TAG);
            if (fragment != null) {
                fragment.loadSelectedLyrics(urls.get(holder.getBindingAdapterPosition()));
            }
        });

    }

    @Override
    public int getItemCount() {
        return titles.size();
    }
}

File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/adapters/FoundLyricsAdapter.kt
package com.atomykcoder.atomykplay.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.atomykcoder.atomykplay.R
import com.atomykcoder.atomykplay.adapters.viewHolders.FoundLyricsViewHolder
import com.atomykcoder.atomykplay.constants.FragmentTags.ADD_LYRICS_FRAGMENT_TAG
import com.atomykcoder.atomykplay.fragments.AddLyricsFragment
import com.atomykcoder.atomykplay.ui.MainActivity

class FoundLyricsAdapter(
    private val titles: ArrayList<String>?,
    private val sampleLyrics: ArrayList<String>?,
    private val urls: ArrayList<String>?,
    _context: Context
) : RecyclerView.Adapter<FoundLyricsViewHolder>() {
    private val mainActivity: MainActivity

    init {
        mainActivity = _context as MainActivity
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FoundLyricsViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.found_lyrics_item, parent, false)
        return FoundLyricsViewHolder(view)
    }

    override fun onBindViewHolder(holder: FoundLyricsViewHolder, position: Int) {
        holder.song_title.text = titles?.get(position)
        holder.song_sampleLyrics.text = sampleLyrics?.get(position)
        holder.itemView.setOnClickListener { view: View? ->
            mainActivity.setBottomSheetState()
            val fragment =
                mainActivity.supportFragmentManager.findFragmentByTag(ADD_LYRICS_FRAGMENT_TAG) as AddLyricsFragment?
            fragment?.loadSelectedLyrics(urls?.get(holder.bindingAdapterPosition))
        }
    }

    override fun getItemCount(): Int {
        return titles?.size ?:0
    }
}
